public/js/postie.js

```diff
@@ -0,0 +1,165 @@
+(function( $ ) {
+
+  var log = {
+    debug: function() { console.log.apply(console, arguments); },
+    info: function() { console.info.apply(console, arguments); },
+    warn: function() { console.warn.apply(console, arguments); },
+    error: function() { console.error.apply(console, arguments); },
+  };
+
+  var $form = $('form.try');
+
+  $('form.try').submit(function( e ) {
+    try {
+      tryRequest( this );
+    } catch (e) {
+      log.error(e);
+    }
+    return false;
+  });
+
+  var $url = $('input[name=url]', $form)
+    .val(localStorage.url)
+    .change( saveToLocalStorage('url') );
+
+  var $method = $('select[name=method]', $form)
+    .val(localStorage.method || 'GET')
+    .change( saveToLocalStorage('method') );
+
+  var $contentType = $('select[name=content-type]', $form)
+    .val(localStorage.contentType || 'GET')
+    .change( saveToLocalStorage('contentType') );
+
+  var $headersRaw = $('textarea[name=headers-raw]', $form)
+    .val(localStorage.headersRaw)
+    .change( saveToLocalStorage('headersRaw') );
+
+  var $payloadRaw = $('textarea[name=payload-raw]', $form)
+    .val(localStorage.payloadRaw)
+    .change( saveToLocalStorage('payloadRaw') );
+
+  var $requestHeaders = $('textarea[name=request-headers]', $form);
+  var $requestRaw = $('textarea[name=request-raw]', $form);
+  var $responseHeaders = $('textarea[name=response-headers]', $form);
+  var $responseRaw = $('textarea[name=response-raw]', $form);
+  var $status = $('.status', $form);
+  var $sampleNodejs = $('textarea[name=sample-nodejs]', $form);
+
+
+  function tryRequest( form ) {
+    log.debug('trying');
+    var url = $url.val();
+    var method = $method.val();
+    var headers = getHeaders();
+    var payload = $payloadRaw.val();
+
+    $requestHeaders.val('');
+    $responseHeaders.val('');
+    $responseRaw.val('');
+    $status.text('In progress...');
+    $form.addClass('request-pending').removeClass('request-failed');
+
+
+    $sampleNodejs.val('');
+    var reqParams = {
+      url: url,
+      method: method,
+      headers: headers,
+      data: payload
+    };
+    resolveTemplate( 'nodejs-request.js', reqParams ).then(function( sample ) {
```

Comment:
Sure, templating is better, this is my 5min templating solution and subject for further refactor once we know we want this ;-)